src/components/Container.js

```diff
@@ -51,19 +52,30 @@ class Container extends React.Component {
     this.drawEnd = this.drawEnd.bind(this);
     this.modifyEnd = this.modifyEnd.bind(this);
     this.state = {
-      drawerOpen: false
+      drawerOpen: false,
+      message: null,
     };
   }
 
 
 
   newTrailClicked = () => {
     const { addTrail } = this.props;
+    this.setState({drawerOpen: true })
     let id = new Date().getTime();
     id = id.toString();
     const name = 'New Trail';
     const trail = new Trail({ id, name, features: [] });
     addTrail(trail);
+    this.setState({
+      message: 'New Trail Added'
+    });
+  }
```

Comment:
You shouldn't call `setState` twice in the same function. Each call will trigger a re-render, just do a single `setState` at the end.

Reply:
Makes sense, I'll take a look